File: Question-One/shopping_list.py
def add_item(shopping_list,name,price,quantity=1):
    for item in shopping_list:
        if item['name'].lower()==name.lower():  
            item['quantity']+=quantity
            print(f"Updated {item['name'] } quantity to {item['quantity']}.")
            return 
            shopping_list.append({'name',name,'price',price,'quantity',quantity})
            print(f'added {name} to the shopping list.')
def remove_item(shopping_list,name):
    for item in shopping_list:
        if item['name'].lower()==name.lower():
            shopping_list.remove(item)
            print(f"Removed '{name}' from the shopping list.")
            return
        print(f"item '{name}' not found in the shopping list.")

def total_cost(shopping_list):
    return sum(item['price']*item['quantity'] for item in shopping_list)

File: Question-One/shopping_list.kt

fun main(){
    val shoppingList = ShoppingList()

    shoppingList.addItem("Milk", 1.99, 2)
    shoppingList.addItem("Bread", 2.49)
    shoppingList.addItem("Eggs", 3.99, 12)
    shoppingList.removeItem("eggs")
    println(shoppingList.totalCost())
}


data class Item(
    val name: String,
    val price: Double,
    var quantity: Int = 1
)

class ShoppingList {
    val items = mutableListOf<Item>()

    fun addItem(name: String, price: Double, quantity: Int = 1) {
        val existingItem = items.find { it.name.equals(name, ignoreCase = true) }
        if (existingItem != null) {
            existingItem.quantity += quantity
            println("Updated ${existingItem.name} quantity to ${existingItem.quantity}.")
        } else {
            items.add(Item(name, price, quantity))
            println("Added ${name} to the shopping list.")
        }
    }

    fun removeItem(name: String) {
        val removed = items.removeIf { it.name.equals(name, ignoreCase = true) }
        if (removed) {
            println("Removed '$name' from the shopping list.")
        } else {
            println("Item '$name' not found.")
        }
    }

    fun totalCost(): Double {
        return items.sumOf { it.price * it.quantity }
    }
}
